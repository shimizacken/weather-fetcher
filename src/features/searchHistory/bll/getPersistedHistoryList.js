import { historyListLocalStorageKey } from 'app/features/searchHistory/bll/historyListStorageKey';

export const sortHistoryList = (historyList) => {
  return historyList?.sort((a, b) => b.date - a.date);
};

export const getPersistedHistoryList = () => {
  const searchHistory = window.localStorage.getItem(historyListLocalStorageKey);

  if (searchHistory) {
    return JSON.parse(searchHistory);
  }
};
