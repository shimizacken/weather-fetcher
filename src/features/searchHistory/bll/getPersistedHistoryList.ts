import {HistoryItem} from './historyListType';
import {historyListLocalStorageKey} from 'app/features/searchHistory';

export const sortHistoryList = (historyList: HistoryItem[]) => {
    return historyList?.sort((a, b) => b.date - a.date);
};

export const getPersistedHistoryList = () => {
    const searchHistory = window.localStorage.getItem(
        historyListLocalStorageKey
    );

    if (searchHistory) {
        return JSON.parse(searchHistory);
    }
};
