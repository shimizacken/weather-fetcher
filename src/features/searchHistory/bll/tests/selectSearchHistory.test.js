import { sortHistoryList } from '../getPersistedHistoryList';

/* eslint-disable no-undef */
describe('selectSearchHistory test suite', () => {
  it('should return undefined when no argument is passed', () => {
    expect(sortHistoryList()).toMatchSnapshot();
  });

  it('should return empty array when historyList is empty', () => {
    const historyList = [];

    expect(sortHistoryList(historyList)).toMatchSnapshot();
  });

  it('should return date descending sorted list', () => {
    const historyList = [
      {
        name: 'Drammen',
        date: 12345
      },
      {
        name: 'Lysaker',
        date: 12343
      },
      {
        name: 'Oslo',
        date: 12344
      }
    ];

    expect(sortHistoryList(historyList)).toMatchSnapshot();
  });
});
