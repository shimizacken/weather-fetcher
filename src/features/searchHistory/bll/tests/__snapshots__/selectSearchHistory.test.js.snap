// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectSearchHistory test suite should return date descending sorted list 1`] = `
Array [
  Object {
    "date": 12345,
    "name": "Drammen",
  },
  Object {
    "date": 12344,
    "name": "Oslo",
  },
  Object {
    "date": 12343,
    "name": "Lysaker",
  },
]
`;

exports[`selectSearchHistory test suite should return empty array when historyList is empty 1`] = `Array []`;

exports[`selectSearchHistory test suite should return undefined when no argument is passed 1`] = `undefined`;
